core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -136,33 +137,42 @@ public ResponsePath doWork(PointList waypoints, List<Path> paths, EncodedValueLo
 
                 fullPoints.add(tmpPoints);
                 responsePath.addPathDetails(PathDetailsFromEdges.calcDetails(path, evLookup, weighting, requestedPathDetails, pathBuilderFactory, origPoints, graph));
+                wayPointIntervals.add(new ResponsePath.Interval(origPoints, pathIndex < paths.size() - 1 ? fullPoints.size() : fullPoints.size() - 1));
                 origPoints = fullPoints.size();
             }
 
             allFound = allFound && path.isFound();
         }
 
-        if (!fullPoints.isEmpty()) {
-            responsePath.addDebugInfo("simplify (" + origPoints + "->" + fullPoints.size() + ")");
```

Comment:
This debug info made no sense anyway, because origPoints was always equal to fullPoints.size()...